webapp/channels/src/components/button/index.tsx

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import classNames from 'classnames';
+import type {ButtonHTMLAttributes} from 'react';
+import React from 'react';
+import type {MessageDescriptor} from 'react-intl';
+import {useIntl} from 'react-intl';
+
+type ButtonProps = {
+    label: string | MessageDescriptor;
+    emphasis: 'primary' | 'secondary' | 'tertiary' | 'quaternary' | 'link';
+    size?: 'xSmall' | 'small' | 'medium' | 'large';
+    pull?: 'left' | 'right';
```

Comment:
Re: Pull
I agree. Location of the component should be handled by the parent component. But that may imply wrapping things in extra divs, which some people don't like (unless we allow for custom styles, and we break the whole approach here). So this is kind of a quick approach. But I am fine moving this out.

Re: Full width
I have mixed feelings here, because one detail we also have is the size of the button, which may affect this a bit. But we would just have to check. But I also agree that if possible, should be the parent reponsibility to make the button full width.

Re: Follow up PR
YES :smile: